plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/PushMojo.java

```diff
@@ -93,21 +98,68 @@
     private OpenShiftBuildStrategy buildStrategy = OpenShiftBuildStrategy.s2i;
 
     @Parameter(property = "docker.skip.push", defaultValue = "false")
-    protected boolean skipPush;
+    private boolean skipPush;
+
+    @Parameter(property = "docker.push.registry")
+    private String pushRegistry;
+
+    @Parameter(property = "docker.source.dir", defaultValue="src/main/docker")
+    private String sourceDirectory;
+
+    @Parameter(property = "docker.target.dir", defaultValue="target/docker")
+    private String outputDirectory;
+
+    @Parameter(property = "fabric8.build.jib", defaultValue = "false")
+    private boolean isJib;
+
+    private static String EMPTY_STRING = "";
+
+    private static String TAR_POSTFIX = ".tar";
 
     @Override
     protected String getLogPrefix() {
         return "F8> ";
     }
 
+    protected boolean isJibMode() {
+        return isJib || Configs.asBoolean(getProperty("fabric8.build.jib"));
+    }
+
+    private String getProperty(String key) {
+        String value = System.getProperty(key);
+        if (value == null) {
+            value = project.getProperties().getProperty(key);
+        }
+        return value;
```

Comment:
This looks like a util function, I think there should already be some method available in EnvUtil for this

Reply:
This method should be inherited from the `AbstractFabric8Mojo` class, but since `BuildMojo` and `PushMojo` don't inherit the `AbstractFabric8Mojo` this method has been defined separately for just these two cases.